plugins/cpu/fu-cpu-device.c

```diff
@@ -382,6 +425,250 @@ fu_cpu_device_add_security_attrs (FuDevice *device, FuSecurityAttrs *attrs)
 	fu_cpu_device_add_security_attrs_intel_smap (self, attrs);
 }
 
+static gboolean
+fu_cpu_exec_cmd (const gchar* command, GError** error)
+{
+	gint exit_status;
+
+	if (!g_spawn_command_line_sync (command, NULL, NULL, &exit_status, error)) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_FAILED,
+			     "failed to execute command: %s", command);
+		return FALSE;
+	}
+
+	if (!g_spawn_check_exit_status(exit_status, error)) {
+		g_set_error_literal (error,
+				     G_IO_ERROR,
+				     G_IO_ERROR_FAILED,
+				     "failed to early load microcode");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+fu_cpu_late_load_microcode (GError** error)
+{
+	char reload_char = '1';
+	int fd;
+	const gchar* late_load_path = "/sys/devices/system/cpu/microcode/reload";
```

Comment:
@hughsie my thought is this should be a configurable path in `fu-common.c` so we can potentially have a test suite file that can be touched to validate all of this, agree?

Reply:
Agree indeed.

Reply:
yes, something exactly like that

Reply:
Usually we get the directory not the file, but don't feel strongly.